pcdsdevices/fms.py

```diff
@@ -0,0 +1,259 @@
+       
+    def __init__(self, *args, elevations = [], **kwargs):
+        self.elevation = elevations
+        super().__init__(*args, **kwargs)
+
+class PDU_Temp4(Device):
+    Sensor1 = FCpt(EpicsSignalRO, '{self.prefix}:{self.elevation[0]}:Sensor:1:GetTempValue', kind='hinted')
+    Sensor2 = FCpt(EpicsSignalRO, '{self.prefix}:{self.elevation[1]}:Sensor:2:GetTempValue', kind='hinted')
+    Sensor3 = FCpt(EpicsSignalRO, '{self.prefix}:{self.elevation[2]}:Sensor:1:GetTempValue', kind='hinted')
+    Sensor4 = FCpt(EpicsSignalRO, '{self.prefix}:{self.elevation[3]}:Sensor:2:GetTempValue', kind='hinted')
+    Sensor5 = FCpt(EpicsSignalRO, '{self.prefix}:{self.elevation[4]}:Sensor:1:GetTempValue', kind='hinted')
+    Sensor6 = FCpt(EpicsSignalRO, '{self.prefix}:{self.elevation[5]}:Sensor:2:GetTempValue', kind='hinted')
+    Sensor7 = FCpt(EpicsSignalRO, '{self.prefix}:{self.elevation[6]}:Sensor:2:GetTempValue', kind='hinted')
+    Sensor8 = FCpt(EpicsSignalRO, '{self.prefix}:{self.elevation[7]}:Sensor:2:GetTempValue', kind='hinted')
+    
+    def __init__(self, *args, elevations = [], **kwargs):
+        self.elevation = elevations
+        super().__init__(*args, **kwargs)
+
+"""
+class PDU_Temp4(Device):
+    Sensor1 = FCpt(EpicsSignalRO, '{self.prefix}:{self.elevation[0]}:Sensor:1:GetTempValue', kind='hinted')
+    Sensor2 = FCpt(EpicsSignalRO, '{self.prefix}:{self.elevation[1]}:Sensor:2:GetTempValue', kind='hinted')
+    Sensor3 = FCpt(EpicsSignalRO, '{self.prefix}:{self.elevation[2]}:Sensor:1:GetTempValue', kind='hinted')
+    Sensor4 = FCpt(EpicsSignalRO, '{self.prefix}:{self.elevation[3]}:Sensor:2:GetTempValue', kind='hinted')
+
+    def __init__(self, *args, elevations = [], **kwargs):
+        self.elevation = elevations
+        super().__init__(*args, **kwargs)
+"""
+#PDU_Temp1(elevations = [15, 15, 32])
+
+class PDU_Humidity1(Device):
+    Sensor1 = FCpt(EpicsSignalRO, '{self.prefix}:{self.elevation[0]}:Sensor:1:GetHumidValue', kind='hinted')
+    Sensor2 = FCpt(EpicsSignalRO, '{self.prefix}:{self.elevation[1]}:Sensor:2:GetHumidValue', kind='hinted')
+    Sensor3 = FCpt(EpicsSignalRO, '{self.prefix}:{self.elevation[2]}:Sensor:1:GetHumidValue', kind='hinted')
+    Sensor4 = FCpt(EpicsSignalRO, '{self.prefix}:{self.elevation[3]}:Sensor:2:GetHumidValue', kind='hinted')
+
+    def __init__(self, *args, elevations = [], **kwargs):
+        self.elevation = elevations
+        super().__init__(*args, **kwargs)
+
+#PDU_Temp2: Will be for portable racks or Racks that have a single PDU, with a total of 2 sensors for monitoring
+class PDU_Humidity2(Device):
+    Sensor1 = FCpt(EpicsSignalRO, '{self.prefix}:{self.elevation[0]}Sensor:1:GetHumidValue', kind='hinted')
+    Sensor2 = FCpt(EpicsSignalRO, '{self.prefix}:{self.elevation[1]}Sensor:2:GetHumidValue', kind='hinted')
+
+    def __init__(self, *args, elevations = [], **kwargs):
+        self.elevation = elevations
+        super().__init__(*args, **kwargs)
+
+class PDU_Humidity3(Device):
```

Comment:
The class numbering scheme here is a bit confusing

Reply:
What aspect of it is confusing/counterintuitive?

Reply:
These are ordered sequentially by where they are placed in the file with no class function meanings.

Reply:
I took the number in the class name to mean the number of components in the class.  Is that right?

Reply:
As in: class 1 has 4 elements, class 2 has 2, class 3 has 6, etc.

Reply:
Yes, you're right that its counterintuitive in that regard for both the Humidity and Temp classes. A solution for that could be switching the class 1 to have 2 elements, class 2 to have 4 elements, and class 3 to have 6? However, I am worried that switching the numbers would mean re-editing the device_class for the majority of the happi entries, which is quite a bit of work. Would this be an important edit?

Reply:
It wouldn't be too much work if you edited the db file directly, it could be as simple as a series of find-replace commands.  I think it's important to try to be as logical as possible with naming, as any small confusions we have now will be exaggerated in the future.  

I can help you poke around the actual database file

Reply:
Ok good to know, thank you! So then what naming scheme would be best for temp and humidity? Humidity1 for 2 entries, Humidity2 for 4, and Humidity3 for 6? Or should we make it so that the number directly corresponds to the number of sensors we are describing in that device class (Humidity2, Humidity4, Humidity6)?

Reply:
This is always a judgement call. I think making them the number of sensors is the most intuitive because then your class name is self-describing instead of relying on the context of the other classes. Either way is OK though because they seem to come in pairs anyway.